lib/spareval/src/lib.rs

```diff
@@ -596,6 +818,155 @@ pub trait AggregateFunctionAccumulator {
     fn finish(&mut self) -> Option<Term>;
 }
 
+/// An extended SPARQL query [dataset specification](https://www.w3.org/TR/sparql11-query/#specifyingDataset).
+///
+/// Allows setting blank node graph names and that the default graph is the union of all named graphs.
+#[derive(Eq, PartialEq, Debug, Clone, Hash)]
+pub struct QueryDatasetSpecification {
+    default: Option<Vec<GraphName>>,
+    named: Option<Vec<NamedOrBlankNode>>,
+}
+
+impl QueryDatasetSpecification {
+    pub fn new() -> Self {
+        Self {
+            default: Some(vec![GraphName::DefaultGraph]),
+            named: None,
+        }
+    }
+
+    /// Checks if this dataset specification is the default one
+    /// (i.e., the default graph is the store default graph, and all named graphs included in the queried store are available)
+    pub fn is_default_dataset(&self) -> bool {
+        self.default
+            .as_ref()
+            .is_some_and(|t| t == &[GraphName::DefaultGraph])
+            && self.named.is_none()
+    }
```

Comment:
```suggestion
    pub fn is_default_dataset(&self) -> bool {
        self == Self::default()
    }
```

simpler.

and shouldn't it be renamed to `is_default_dataset_specifically`? (or `is_default` if you want it to be short)

Reply:
> simpler

But requires an alloc

> `is_default`

Good idea. Will add a TODO (I just moved this struct so I don't want to introduce breaking changes).

Reply:
Hmm I assumed the compiler would be smart enough to compile it to the same code, but I just checked it's not. I guess I have Rust code elsewhere to update, then...